torch/distributed/_shard/sharding_spec/chunk_sharding_spec_ops/embedding.py

```diff
@@ -39,34 +41,25 @@ def sharded_embedding(types, args, kwargs, pg):
     4 GPUs creating 3 shard of (3 x 17) and 1 shard of (1 x 17).
     The algorithm is as follows:
 
-    1. First the input is flattened to 1D and gets sorted so that we can distribute
-       them to the corresponding rank. For example if the given input is
+    1. First the input is broadcasted to all ranks, since this is SPMD we
```

Comment:
should this be an all_gather or broadcast, looks confusing to me, usually if input is sharded, we all_gather, if input is not, we broadcast it to get the same values, which one we are taking here?